atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -148,4 +149,15 @@ public long getSweepPauseMillis() {
     public int getSweepBatchSize() {
         return AtlasDbConstants.DEFAULT_SWEEP_BATCH_SIZE;
     }
+
+    @Value.Check
+    protected final void check() {
+        Preconditions.checkState(!(leader().isPresent() && lock().isPresent()),
+                "Leader and lock configuration blocks must not both be present.");
```

Comment:
Having read this again, I think our intent is clearer with statements (and checks) like

> "If a leader configuration block is present, lock and timestamp configuration blocks must not be present"

and

> "If the leader configuration block is absent, both the lock and timestamp server blocks must be present"